collect_app/src/main/java/org/odk/collect/android/utilities/SharedPreferencesUtils.java

```diff
@@ -117,37 +119,15 @@ public void savePreferencesFromJSON(JSONObject settingsJson) throws JSONExceptio
             }
         }
 
-        checkQRCodeForRemovedSettings(generalPrefsJson);
+        // check and migrate if auto send preference needs to be migrated
+        if (generalPrefsJson.has(KEY_AUTOSEND_WIFI) || generalPrefsJson.has(KEY_AUTOSEND_NETWORK)) {
```

Comment:
Same as with the settings file, I don't think you need these extra checks, you can do them in the migrate method.